bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2854,6 +2854,13 @@ public boolean isDisposed () {
 	return device == null;
 }
 
+private static char replaceBelowSpaceCharacters(char chr) {
+	return chr >= ' '
+		? chr
+		//: (char)(0x2400 + chr);
+		: '?';
```

Comment:
Given that \t\r\n all fall into this category, can this really be correct?

Reply:
You are right. The bug was not directly visible, but caused weird effects when editing.